extensions/wikia/InsightsV2/InsightsLoopController.class.php

```diff
@@ -0,0 +1,203 @@
+		}
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shownwhen user dont fix an issue
+	 *
+	 * @param string $subpage Insights subpage
+	 * @param Title $title
+	 * @param InsightsQueryPageModel $model
+	 * @return array
+	 */
+	private function getNotFixedNotificationParams( $subpage, Title $title, InsightsQueryPageModel $model ) {
+		$params = $this->getInsightListLinkParams( $subpage );
+
+		if ( $model->getConfig()->displayFixItMessage() ) {
+			$params = array_merge( $params, $this->getInsightFixItParams( $title, $model ) );
+		}
+
+		$params['notificationMessage'] = wfMessage( InsightsHelper::INSIGHT_INPROGRESS_MSG_PREFIX . $subpage )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shown when user fix all issues in given insight type
+	 */
+	private function getCongratulationsNotificationParams() {
+		$params = $this->getInsightLinkParams();
+		$params['notificationMessage'] = wfMessage( 'insights-notification-message-alldone' )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shown when user fix one issue from the insights list
+	 *
+	 * @param string $subpage Insights subpage
+	 * @param array $next An array with a url and text for a link to the next item
+	 * @return array
+	 */
+	private function getInsightFixedNotificationParams( $subpage, $next ) {
+		$params = $this->getInsightNextLinkParams( $subpage, $next );
+		$params = array_merge( $params, $this->getInsightListLinkParams( $subpage ));
+		$params['notificationMessage'] = wfMessage( InsightsHelper::INSIGHT_FIXED_MSG_PREFIX . $subpage )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params to generate next item link in notification template
+	 *
+	 * @param String $subpage Insights subpage
+	 * @param array $next An array with a url and text for a link to the next item
+	 * @return array
+	 */
+	private function getInsightNextLinkParams( $subpage, $next ) {
+		return [
+			'nextArticleText' => wfMessage( InsightsHelper::INSIGHT_NEXT_MSG_PREFIX . $subpage )->plain(),
+			'nextArticleTitle' => $next['link']['text'],
+			'nextArticleLink' => $next['link']['url']
+		];
+	}
+
+	/**
+	 * Get params to generate link back to edit mode
+	 *
+	 * @param Title $title
+	 * @param InsightsQueryPageModel $model
+	 * @return array
+	 */
+	private function getInsightFixItParams( Title $title, InsightsQueryPageModel $model ) {
+		$link = InsightsHelper::getTitleLink( $title, $model->getUrlParams() );
+
+		return [
+			'editPageText' => wfMessage( 'insights-notification-message-fixit' )->plain(),
+			'editPageLink' => $link['url']
+		];
+	}
+
+	/**
+	 * Get params to generate link to insight main page in notification template
+	 */
+	private function getInsightLinkParams() {
+		return [
+			'insightsPageText' => wfMessage( 'insights-notification-see-more' )->plain(),
+			'insightsPageLink' => SpecialPage::getTitleFor( 'Insights' )
```

Comment:
It returns Title object and previously here was string, so probably you're missing ``SpecialPage::getTitleFor(...)->getLocalUrl()`` here.